kgraphql/src/main/kotlin/com/apurebase/kgraphql/schema/SchemaPrinter.kt

```diff
@@ -0,0 +1,290 @@
+        }
+
+        // Enums
+        //  https://spec.graphql.org/draft/#sec-Enums
+        val enums = buildString {
+            schemaTypes.filter { it.kind == TypeKind.ENUM }.forEachIndexed { index, type ->
+                if (index > 0) {
+                    appendLine()
+                }
+                appendDescription(type)
+                appendLine("enum ${type.name} {")
+                val indentation = "  "
+                type.enumValues.sortedByName().forEach { enumValue ->
+                    appendDescription(enumValue, indentation)
+                    appendLine("${indentation}${enumValue.name}${enumValue.deprecationInfo()}")
+                }
+                appendLine("}")
+            }
+        }
+
+        // Input Types
+        //  https://spec.graphql.org/draft/#sec-Input-Objects
+        val inputTypes = buildString {
+            schemaTypes.filter { it.kind == TypeKind.INPUT_OBJECT && !it.inputFields.isNullOrEmpty() }
+                .forEachIndexed { index, type ->
+                    if (index > 0) {
+                        appendLine()
+                    }
+                    appendDescription(type)
+                    appendLine("input ${type.name}${type.implements()} {")
+                    val indentation = "  "
+                    type.inputFields.sortedByName().forEach { field ->
+                        appendDescription(field, indentation)
+                        appendLine("${indentation}${field.name}: ${field.type.typeReference()}${field.deprecationInfo()}")
+                    }
+                    appendLine("}")
+                }
+        }
+
+        // Directives
+        //   https://spec.graphql.org/draft/#sec-Type-System.Directives.Built-in-Directives
+        //   "When representing a GraphQL schema using the type system definition language any built-in directive may be omitted for brevity."
+        val directives = buildString {
+            schema.directives.filter { config.includeBuiltInDirectives || !it.isBuiltIn() }.sortedByName()
+                .forEachIndexed { index, directive ->
+                    if (index > 0) {
+                        appendLine()
+                    }
+                    val args = directive.args.takeIf { it.isNotEmpty() }
+                        ?.joinToString(", ", prefix = "(", postfix = ")") { arg ->
+                            arg.name + ": " + arg.type.typeReference() + arg.defaultInfo()
+                        } ?: ""
+                    appendLine("directive @${directive.name}$args on ${directive.locations.joinToString(" | ") { it.name }}")
+                }
+        }
+
+        return listOf(
+            schemaDefinition,
+            scalars,
+            objects,
+            interfaces,
+            unions,
+            enums,
+            inputTypes,
+            directives
+        ).filterNot { it.isBlank() }.joinToString("\n")
+    }
+
+    private fun includeSchemaDefinition(schema: __Schema): Boolean =
```

Comment:
This is quite complex, I think an explanation here would help to grasp why this is how it is

Reply:
There is a comment but apparently in the wrong place then 😉 Will update.